src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -3483,6 +3483,22 @@ private void EmitConditionalOperator(BoundConditionalOperator expr, bool used)
         {
             Debug.Assert(expr.ConstantValueOpt == null, "Constant value should have been emitted directly");
 
+            // Generate branchless IL for (b ? 1 : 0).
+            if (used && _ilEmitStyle != ILEmitStyle.Debug &&
+                (IsNumeric(expr.Type) || expr.Type.PrimitiveTypeCode == Cci.PrimitiveTypeCode.Boolean) &&
```

Comment:
>IsNumeric(expr.Type) [](http://example.com/codeflow?start=17&length=20)

Do we care if IntPtr/UIntPtr are treated as native integers?

Reply:
We probably don't, but it seems it's not possible to have an `IntPtr` constant which is not an `nint`.